app/src/main/java/com/kms/cura/view/CheckBoxAdapter.java

```diff
@@ -0,0 +1,98 @@
+package com.kms.cura.view;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.CheckBox;
+import android.widget.LinearLayout;
+import android.widget.SpinnerAdapter;
+import android.widget.TextView;
+
+import com.kms.cura.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by toanbnguyen on 5/30/2016.
+ */
+public class CheckBoxAdapter extends BaseAdapter {
+    Context context;
+    int resource;
+    ArrayList<String> checkList;
+    boolean[] checked;
+    public CheckBoxAdapter(Context context, int resource, ArrayList objects, boolean[] checked) {
+        this.context = context;
+        this.resource = resource;
+        this.checkList = objects;
+        if (checked != null)
+            this.checked = checked;
+        else {
+            this.checked = new boolean[checkList.size() - 1];
+            for (int i = 0; i < checkList.size() - 1; ++i)
+                this.checked[i] = false;
+        }
+    }
+
+    @Override
+    public int getCount() {
+        return checkList.size() - 1; // not to include the hint in the list
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return checkList.get(position);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return position;
+    }
+
+    @Override
+    public View getView(final int position, View convertView, ViewGroup parent) {
```

Comment:
I cannot figure out an idea to separate this, everything is related to each others, if there's another function, it must require too many parameters. I will put some comments for the flow of this function